src/components/bcros/filing/CommonTemplate.vue

```diff
@@ -109,9 +110,16 @@ const isStatusApproved = computed(() => isFilingStatus(filing.value, FilingStatu
 
 const isShowBody = ref(false)
 
-const showDetails = () => {
+const showDetails = async () => {
   if (filing.value.documents === undefined && filing.value.documentsLink) {
-    loadDocumentList(filing.value)
+    ui.fetchingData = true
+
+    await loadDocumentList(filing.value)
+
+    // wait for another 500ms to show the loading modal
```

Comment:
This looks like a port of the code from Filings UI. I'd like to see a better explanation of why you have the "sleep" here, ie:

![image](https://github.com/user-attachments/assets/267aa94a-3902-4c9f-818f-0cb1540e710a)